.github/workflows/ci.yml

```diff
@@ -0,0 +1,333 @@
+  pull_request:
+    branches:
+      - 'main'
+  workflow_dispatch:
+env:
+  docker_registry: 'us-docker.pkg.dev'
+  docker_tag: '${{ github.sha }}'
+  docker_repo: 'us-docker.pkg.dev/lumberjack-dev-infra/images'
+  wif_provider: 'projects/638387980668/locations/global/workloadIdentityPools/github-pool/providers/github-provider'
+  wif_service_account: 'gh-access-sa@lumberjack-dev-infra.iam.gserviceaccount.com'
+  server_project_id: 'github-ci-server'
+  client_project_id: 'github-ci-app-0'
+  bigquery_dataset_id: 'audit_logs.audit_abcxyz_data_access'
+concurrency:
+  group: '${{ github.workflow }}-${{ github.head_ref || github.ref }}'
+  cancel-in-progress: true
+jobs:
+  # Linting jobs - terraform, java, go
+  terraform_lint:
+    uses: 'abcxyz/pkg/.github/workflows/terraform-lint.yml@cc2b918b16df0f627b933ed8dddf0d846fa9108b' # ratchet:abcxyz/pkg/.github/workflows/terraform-lint.yml@main
+    with:
+      directory: 'terraform'
+      terraform_version: '1.2'
+  go_lint_cmd:
+    uses: 'abcxyz/pkg/.github/workflows/go-lint.yml@cc2b918b16df0f627b933ed8dddf0d846fa9108b' # ratchet:abcxyz/pkg/.github/workflows/go-lint.yml@main
+    with:
+      directory: 'cmd'
+      go_version: '1.19'
+  go_lint_pkg:
+    uses: 'abcxyz/pkg/.github/workflows/go-lint.yml@cc2b918b16df0f627b933ed8dddf0d846fa9108b' # ratchet:abcxyz/pkg/.github/workflows/go-lint.yml@main
+    with:
+      directory: 'pkg'
+      go_version: '1.19'
+  go_lint_integration:
+    uses: 'abcxyz/pkg/.github/workflows/go-lint.yml@cc2b918b16df0f627b933ed8dddf0d846fa9108b' # ratchet:abcxyz/pkg/.github/workflows/go-lint.yml@main
+    with:
+      directory: 'integration'
+      go_version: '1.19'
+  java_lint:
+    uses: 'abcxyz/pkg/.github/workflows/java-lint.yml@cc2b918b16df0f627b933ed8dddf0d846fa9108b' # ratchet:abcxyz/pkg/.github/workflows/java-lint.yml@main
+    with:
+      directory: 'clients/java-logger'
+      java_version: '11'
+  # Unit tests - java, go
+  go_test:
+    uses: 'abcxyz/pkg/.github/workflows/go-test.yml@cc2b918b16df0f627b933ed8dddf0d846fa9108b' # ratchet:abcxyz/pkg/.github/workflows/go-test.yml@main
+    with:
+      go_version: '1.19'
+      go_packages: |-
+        $(go list ./... | grep -v -E 'lumberjack/third_party|lumberjack/integration')
+  java_test:
+    runs-on: 'ubuntu-latest'
+    steps:
+      - uses: 'actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8' # ratchet:actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@a18c333f3f14249953dab3e186e5e21bf3390f1d # ratchet:actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Run tests with Maven
+        run: |-
+          mvn clean test --no-transfer-progress -f clients/java-logger
+
+  # Build the main lumberjack server and push to artifact registry
+  build-lumberjack-server:
+    runs-on: 'ubuntu-latest'
+    needs: [terraform_lint, go_lint_cmd, go_lint_pkg, go_lint_integration, java_lint, go_test, java_test]
```

Comment:
If you wanted to make this a little more repeatable, you could create a noop job:

```yaml
jobs:
  go_test: 
    # ...
  java_test:
    # ...
  lint_and_unit:
    needs: [terraform_lint, go_lint, java_lint, go_test, java_test]
    steps:
      - run: 'echo ok'
```

And then just `needs: lint_and_unit` in all the other places.